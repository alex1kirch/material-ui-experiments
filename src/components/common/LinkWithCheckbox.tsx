import * as React from "react";
import Link from "./Link";

export interface ILinkWithCheckboxProps {
    isChecked: boolean;
    onChange: (ev: React.ChangeEvent<HTMLInputElement>) => void;
}
export default class LinkWithCheckbox extends React.Component<ILinkWithCheckboxProps> {
    public render() {
        const { isChecked, onChange } = this.props;

        return (
            <Link>
                <input data-test="input" type="checkbox" checked={isChecked} onChange={onChange} />
            </Link>
        );
    }
}
