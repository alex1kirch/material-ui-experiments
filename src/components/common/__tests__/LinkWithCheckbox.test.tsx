import { shallow } from "enzyme";
import enzymeToJson from "enzyme-to-json";
import * as React from "react";
import LinkWithCheckbox, { ILinkWithCheckboxProps } from "../LinkWithCheckbox";

const setup = (propOverrides?: Partial<ILinkWithCheckboxProps>) => {
    const props: ILinkWithCheckboxProps = {
        isChecked: false,
        onChange: jest.fn(),
        ...propOverrides,
    };

    const wrapper = shallow(<LinkWithCheckbox {...props} />);

    return {
        input: wrapper.find("[data-test='input']"),
        props,
        wrapper,
    };
};

describe("<LinkWithCheckbox />", () => {
    it("renders a snapshot", () => {
        const { wrapper } = setup();
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it("should be checked by default", () => {
        const { input } = setup({ isChecked: true });

        const actual = input.props().checked;
        expect(actual).toBeTruthy();
        // expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it("should call onChange on input onChange action", () => {
        const { input, props } = setup();
        input.simulate("change");
        expect(props.onChange).toBeCalled();
    });
});
