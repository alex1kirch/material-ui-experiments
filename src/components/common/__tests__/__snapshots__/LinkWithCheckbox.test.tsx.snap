// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkWithCheckbox /> renders a snapshot 1`] = `
<Link>
  <input
    checked={false}
    data-test="input"
    onChange={[MockFunction]}
    type="checkbox"
  />
</Link>
`;
